src/API/controllers/issue.controller.ts

```diff
@@ -0,0 +1,358 @@
+import { Request, Response } from "express";
+import { IssueService } from "../../app/services/issue.service";
+import { CreateIssueDto, UpdateIssueDto } from "../../domain/DTOs/issueDTO";
+import { validate } from "class-validator";
+import { plainToClass } from "class-transformer";
+import { injectable, inject } from "tsyringe"; // Added inject
+
+@injectable() // Added injectable decorator
+export class IssueController {
+  constructor(@inject(IssueService) private issueService: IssueService) {}
+
+  
+  async create(req: Request, res: Response): Promise<void> { // Renamed from createIssue
+    try {
+      const { projectId } = req.params;
+      const userId = req.user?.id;
```

Comment:
you should access token payload from the body. 
We should add the user object to the request later on as @ahmad-abdelraheem recommend
```suggestion
      const { userId } = req.body;
```